var/spack/repos/builtin/packages/xrootd/package.py

```diff
@@ -60,7 +62,7 @@ class Xrootd(CMakePackage):
     depends_on('cmake@2.6:', type='build')
     depends_on('libxml2', when='+http')
     depends_on('uuid', when="@4.11.0:")
-    depends_on('openssl')
+    depends_on('openssl@1')
```

Comment:
Can we turn this into a range bounded from above? Asking since most packages don't use `@1` like this and since the behavior is different when used from cli:
```console
$ spack spec xrootd ^openssl@1
Input spec
--------------------------------
xrootd
    ^openssl@1

Concretized
--------------------------------
xrootd@5.1.0%gcc@9.3.0+http~ipo~python+readline build_type=RelWithDebInfo cxxstd=11 arch=linux-ubuntu20.04-icelake
    ^bzip2@1.0.8%gcc@9.3.0~debug~pic+shared arch=linux-ubuntu20.04-icelake
        ^diffutils@3.8%gcc@9.3.0 arch=linux-ubuntu20.04-icelake
            ^libiconv@1.16%gcc@9.3.0 libs=shared,static arch=linux-ubuntu20.04-icelake
    ^cmake@3.21.3%gcc@9.3.0~doc+ncurses+openssl+ownlibs~qt build_type=Release arch=linux-ubuntu20.04-icelake
        ^ncurses@6.2%gcc@9.3.0~symlinks+termlib abi=none arch=linux-ubuntu20.04-icelake
            ^pkgconf@1.8.0%gcc@9.3.0 arch=linux-ubuntu20.04-icelake
        ^openssl@1%gcc@9.3.0~docs certs=system arch=linux-ubuntu20.04-icelake
            ^perl@5.34.0%gcc@9.3.0+cpanm+shared+threads arch=linux-ubuntu20.04-icelake
                ^berkeley-db@18.1.40%gcc@9.3.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-ubuntu20.04-icelake
                ^gdbm@1.19%gcc@9.3.0 arch=linux-ubuntu20.04-icelake
                    ^readline@8.1%gcc@9.3.0 arch=linux-ubuntu20.04-icelake
                ^zlib@1.2.11%gcc@9.3.0+optimize+pic+shared arch=linux-ubuntu20.04-icelake
    ^libxml2@2.9.12%gcc@9.3.0~python arch=linux-ubuntu20.04-icelake
        ^xz@5.2.5%gcc@9.3.0~pic libs=shared,static arch=linux-ubuntu20.04-icelake
    ^util-linux-uuid@2.36.2%gcc@9.3.0 arch=linux-ubuntu20.04-icelake
```
while a range has the same behavior in a directive and on the command line.

Reply:
Thanks for catching this and sorry for the oversight! Pushed update to `'openssl@:1'`